packages/backend/src/controllers/house/house.controller.ts

```diff
@@ -46,19 +48,30 @@ export class HouseController {
     @User() user: DecodedIdToken,
   ): Promise<HouseResponseDto> {
     createHouseDto.code = this.houseUtil.generateString(8);
-    createHouseDto.owner = (
-      await this.userStoreService.findOneByFirebaseId(user.uid)
-    )._id;
+    const owner = await this.userStoreService.findOneByFirebaseId(user.uid);
+    createHouseDto.owner = owner._id;
+    const userList: Array<Types.ObjectId> = [];
+    userList.push(owner._id);
     const house = await this.houseStoreService.create(createHouseDto);
     await this.userStoreService.updateByFirebaseId(user.uid, {
       house: house._id,
     });
+    await this.houseStoreService.update(house.id, {
+      users: userList,
+    });
```

Comment:
Is there a reason why the house document is being updated immediately after it is being created? You could just add the usersList to the createHouseDto before you call create on the houseStoreService. For example, replacing lines 51 - 61 with the following:

`const owner = await this.userStoreService.findOneByFirebaseId(user.uid);
createHouseDto.owner = owner._id;

const userList: Array<Types.ObjectId> = [];
userList.push(owner._id);
createHouseDto.users = userList;

const house = await this.houseStoreService.create(createHouseDto);
await this.userStoreService.updateByFirebaseId(user.uid, {
      house: house._id,
 });`

Reply:
Thanks for the advice! I actually tried this before but had some undefined error messages. And I just realized I simply did not declare the array type properly... Will update

Reply:
the createHouseDTO should not expose the users list. But @mped822 is correct in that you shouldn't be updating after creation. Instead have a look at my suggestion to create the house